src/platform/silabs/wifi/icd/WifiSleepManager.cpp

```diff
@@ -150,17 +107,36 @@ CHIP_ERROR WifiSleepManager::VerifyAndTransitionToLowPowerMode(PowerEvent event)
         return CHIP_NO_ERROR;
     }
 
-    if (!WifiInterface::GetInstance().IsWifiProvisioned())
+    if (!mWifiStateProvider->IsWifiProvisioned())
     {
         return ConfigureDeepSleep();
     }
 
     return ConfigureDTIMBasedSleep();
+}
+
+CHIP_ERROR WifiSleepManager::ConfigureDTIMBasedSleep()
+{
+    ReturnLogErrorOnFailure(mPowerSaveInterface->ConfigureBroadcastFilter(false));
```

Comment:
add check for mPowerSaveInterface == nullptr

```suggestion
    VerifyOrDieWithMsg(mPowerSaveInterface != nullptr, DeviceLayer, "WifiSleepManager is not initialized");
    ReturnLogErrorOnFailure(mPowerSaveInterface->ConfigureBroadcastFilter(false));
```

Reply:
I didn't add the check in the private function since they are only called from public function that have the check. 
Should we add the check everywhere?

Reply:
I think it would be safer. If for some reason this ever gets called without proper init it would cause a crash.

Reply:
done